pkg/kubefedctl/federate/federate.go

```diff
@@ -83,17 +84,35 @@ type federateResource struct {
 	outputYAML        bool
 	enableType        bool
 	federateContents  bool
+	filename          string
 }
 
 func (j *federateResource) Bind(flags *pflag.FlagSet) {
 	flags.StringVarP(&j.resourceNamespace, "namespace", "n", "default", "The namespace of the resource to federate.")
 	flags.StringVarP(&j.output, "output", "o", "", "If provided, the resource that would be created in the API by the command is instead output to stdout in the provided format.  Valid format is ['yaml'].")
 	flags.BoolVarP(&j.enableType, "enable-type", "e", false, "If true, attempt to enable federation of the API type of the resource before creating the federated resource.")
 	flags.BoolVarP(&j.federateContents, "contents", "c", false, "Applicable only to namespaces. If provided, the command will federate all resources within the namespace after federating the namespace.")
+	flags.StringVarP(&j.filename, "filename", "f", "", "If specified, the provided yaml file will be used as the input for target resources to federate. Only --output will be accepted from command line with this option. Other flag options if provided will be ignored")
```

Comment:
Is there a reason to support anything but yaml output when this flag is used?  If a user wanted to apply it to a server, they could pipe it to `kubectl apply`.  I'm pretty sure we already had this conversation.

Reply:
I think output to yaml would be preferable, and there would be no confusion if the help text explained that the flag output to yaml.  `kubectl apply` exists and I don't think it's advantageous for us to duplicate its behavior (see: the best code is no code).

Reply:
Updated!